org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/corext/template/java/PostfixTemplate.java

```diff
@@ -85,6 +89,7 @@ class PostfixPreferences {
 	public static final String WHILE_ID = "org.eclipse.jdt.postfixcompletion.while";
 
 	// Default Contents
+	public static final String ASSERT_CONTENT = "assert ${i:inner_expression(java.lang.Boolean)};";
```

Comment:
I haven't really complained about previous postfix contributions that do this because it was a corner case but I wonder if we should address it here :

This will work for `Boolean` objects but not `boolean` primitives. So things like `boolean foo = ...` would not receive this postfix for `foo.as|` . I think if you change this to `inner_expression(boolean,java.lang.Boolean)` it should handle both.